src/peglib.h

```diff
@@ -2603,6 +2604,16 @@ class parser
         return (*grammar_)[s];
     }
 
+    std::vector<std::string> get_rule_names(){
+        std::vector<std::string> rules(grammar_->size());
```

Comment:
In order to avoid allocating a lot of empty strings when creating this vector, it would be more efficient to use

    std::vector<std::string> rules;
    rules.reserve(grammar_->size());
    for (auto const& r : *grammar_) {
        rules.emplace_back(r.first);
    }